File: examples/golang/data/initial/hair_style.go
// +build test

package initial

import (
	"forcedata/examples/golang/model"
)

// データの識別子
const (
	Test01HairStyle1 model.DataID = "Test01HairStyle1"
)

func RegisterHairStyle() {
	f := model.Factory
	f.Register(Test01HairStyle1, func() interface{} {
		return &model.HairStyle{}
	})
}

File: examples/typescript/data/initial/hair_style.ts
import { factory } from '../../model/factory';
import { HairStyle } from '../../model/model';
import { DATAID } from '../dataid';

export function registerInitialHairStyle() {
  const f = factory;
  f.register(DATAID.TEST01_HAIR_STYLE_1, () => {
    return {} as HairStyle;
  });
}
